### PR TITLE
[MM-32393] Add documentation for the custom nginx template

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Adds documentation for the custom `nginx` template for Omnibus.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32393

### CODE REVIEW

**Review by grundleborg — APPROVED** [review]

File: source/install/mattermost-omnibus.rst [hunk 1]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
 - ``data_directory``: This is the directory where Mattermost stores its data.
 - ``enable_plugin_uploads``: This setting can be ``true`` or ``false`` and is used to configure the ``PluginSettings.EnableUploads`` Mattermost configuration property.
 - ``enable_local_mode``: This setting can be ``true`` or ``false`` and is used to configure the ``ServiceSettings.EnableLocalMode`` Mattermost configuration property.
+- ``nginx_template``: Optional path to a custom nginx template.
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:67]:
```suggestion
- ``nginx_template``: Optional path to a custom NGINX template.
```

File: source/install/mattermost-omnibus.rst [hunk 2]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
...
 
 After modifying the ``mmomni.yml`` configuration file, you need to run ``mmomni reconfigure`` for Omnibus to apply the changes and restart Mattermost.
 
+Using a custom nginx template
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:71]:
```suggestion
Using a custom NGINX template
```

File: source/install/mattermost-omnibus.rst [hunk 3]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Mattermost Omnibus generates a ``nginx`` configuration depending on how the different properties of the ``mmomni.yml`` file are set, but it is possible that you need to customize the configuration further to support other use cases, like using custom SSL certificates, for example. For those cases, Omnibus supports using a custom ``nginx`` template to generate its configuration. To use this feature, you need to copy and modify the original template located at ``/opt/mattermost/mmomni/ansible/playbooks/mattermost.conf`` to a new location and then customize it to your needs. You can either use the variables and internal logic that is already bundled in the template, modifying the parts that you need, or just use a fully static configuration instead.
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:74]:
```suggestion
Mattermost Omnibus generates an ``nginx`` configuration depending on how the different properties of the ``mmomni.yml`` file are set. However, you may need to customize the configuration further to support other use cases, such as using custom SSL certificates. For those cases, Omnibus supports using a custom ``nginx`` template to generate its configuration. 

To use this feature, you need to copy and modify the original template located at ``/opt/mattermost/mmomni/ansible/playbooks/mattermost.conf`` to a new location. Then, you can either use the variables and internal logic already bundled in the template and modify the parts that you need, or use a fully static configuration instead.
```

File: source/install/mattermost-omnibus.rst [hunk 4]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
...
+
+After the template has been customized, you need to add a ``nginx_template`` property to the ``/etc/mattermost/mmomni.yml`` configuration file, and then run ``mmomni reconfigure``. The reconfigure process will use the new template to generate the nginx final configuration, and you can check the contents of the ``/etc/nginx/conf.d/mattermost.conf`` file to validate that the changes were applied successfully.
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:76]:
```suggestion
After the template has been customized, add an ``nginx_template`` property to the ``/etc/mattermost/mmomni.yml`` configuration file, and then run ``mmomni reconfigure``. The reconfigure process will use the new template to generate the NGINX final configuration. You can check the contents of the ``/etc/nginx/conf.d/mattermost.conf`` file to validate that the changes were applied successfully.
```

File: source/install/mattermost-omnibus.rst [hunk 5]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
...
+
+Please be careful when using this feature, as making changes to the custom template can cause the reconfigure process to fail, or the generated nginx configuration to be invalid.
+
+This feature is available since Mattermost Omnibus version 5.32.0.
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:80]:
```suggestion
This feature is available from Mattermost Omnibus version 5.32.0.
```

File: source/install/mattermost-omnibus.rst [hunk 6]
```diff
@@ -64,9 +64,21 @@ The properties that you can configure in this file are:
...
+
+After the template has been customized, you need to add a ``nginx_template`` property to the ``/etc/mattermost/mmomni.yml`` configuration file, and then run ``mmomni reconfigure``. The reconfigure process will use the new template to generate the nginx final configuration, and you can check the contents of the ``/etc/nginx/conf.d/mattermost.conf`` file to validate that the changes were applied successfully.
+
+Please be careful when using this feature, as making changes to the custom template can cause the reconfigure process to fail, or the generated nginx configuration to be invalid.
```
Comment (justinegeffen) [inline source/install/mattermost-omnibus.rst:78]:
```suggestion
Please be careful when using this feature, as making changes to the custom template can cause the reconfigure process to fail, or the generated NGINX configuration to be invalid.
```

**Review by justinegeffen — CHANGES_REQUESTED** [review]
Thanks Miguel! This is awesome. I've made some suggestions around wording - let me know if the context has changed at all or if the suggestions don't work. :)

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**